processing/src/main/java/org/apache/druid/query/search/UseIndexesStrategy.java

```diff
@@ -134,10 +134,7 @@ private UseIndexesStrategy(
       }
     }
 
-    return new Pair<List<DimensionSpec>, List<DimensionSpec>>(
-        ImmutableList.copyOf(bitmapDims),
-        ImmutableList.copyOf(nonBitmapDims)
-    );
+    return new Pair<>(bitmapDims, nonBitmapDims);
```

Comment:
Please document, via javadoc, that this method returns editable lists that the caller is free to edit. That's not typical; callers should generally assume that they should _not_ edit the list that comes back from a method, since it might be "owned" by someone else.